addons/html_editor/static/src/core/dom_plugin.js

```diff
@@ -275,8 +279,10 @@ export class DomPlugin extends Plugin {
                         );
                         currentNode = insertBefore ? right : left;
                         const otherNode = insertBefore ? left : right;
-                        if (isBlock(otherNode)) {
+                        if (isUnbreakable(nodeToInsert) && container.childNodes.length === 1) {
```

Comment:
The concept of "unbreakable" has been split into two different concepts in https://github.com/odoo/odoo/pull/174272.